src/app/components/core/configuration-manager/view-config-item/view-config-item.component.html

```diff
@@ -17,31 +17,38 @@
+                    <input *ngIf="getConfigAttributeType(configItem.value.type) === 'TEXT'" name="{{configItem.value.key}}" class="input is-fullwidth is-small"
+                      type="text" [(ngModel)]="configItem.value.value" [value]='configItem?.value?.value != undefined ? configItem?.value?.value : ""'
+                      required [attr.maxLength]="configItem?.value?.length" />
 
-                  <textarea rows="3" *ngIf="getConfigAttributeType(configItem.value.type) === 'LONG_TEXT'" class="textarea is-fullwidth is-small"
-                    name="{{configItem.value.key}}" [(ngModel)]="configItem.value.value" [value]='configItem?.value?.value != undefined ? configItem?.value?.value : ""'
-                    type="text" required>{{configItem?.value?.value}}</textarea>
+                    <input #numField="ngModel" *ngIf="getConfigAttributeType(configItem.value.type) === 'NUMBER'" name="{{configItem.value.key}}"
+                      type="number" [value]='configItem?.value?.value != undefined ? configItem?.value?.value : ""' class="input is-fullwidth is-small"
+                      [min]="configItem?.value?.minimum" [max]="configItem?.value?.maximum" [(ngModel)]="configItem.value.value"
+                      appNumberOnly required (input)="validRange = inputValidator($event)" />
 
-                  <div *ngIf="getConfigAttributeType(configItem.value.type) == 'ENUMERATION'" class="select is-small">
-                    <select name='{{configItem.value.key}}' [(ngModel)]="configItem.value.value">
-                      <option *ngFor="let configOption of configItem.value.options" [value]='configOption'>{{configOption}}</option>
-                    </select>
+                    <textarea rows="3" *ngIf="getConfigAttributeType(configItem.value.type) === 'LONG_TEXT'" class="textarea is-fullwidth is-small"
+                      name="{{configItem.value.key}}" [(ngModel)]="configItem.value.value" [value]='configItem?.value?.value != undefined ? configItem?.value?.value : ""'
+                      type="text" [attr.maxLength]="configItem?.value?.length" required>{{configItem?.value?.value}}</textarea>
+
+                    <div *ngIf="getConfigAttributeType(configItem.value.type) == 'ENUMERATION'" class="select is-small">
+                      <select name='{{configItem.value.key}}' [(ngModel)]="configItem.value.value">
+                        <option *ngFor="let configOption of configItem.value.options" [value]='configOption'>{{configOption}}</option>
+                      </select>
+                    </div>
                   </div>
                 </div>
               </div>
             </div>
           </div>
         </ng-container>
-        &nbsp;
         <div class="field">
           <div *ngIf="!f.valid" class="error">
             <sup>*</sup> Field can't left blank.
           </div>
+          <div *ngIf="!validRange" class="error">
+            <sup>*</sup> Field size out of range.
```

Comment:
what is that range? if we are throwing this error  message, we should know `Value should be in range <min>-<max>`